toucan_connectors/http_api/pagination_configs.py

```diff
@@ -0,0 +1,128 @@
+from enum import Enum
+from typing import Any, Optional
+
+from pydantic import BaseModel, Field
+
+from toucan_connectors.common import FilterSchema
+from toucan_connectors.http_api.http_api_data_souce import HttpAPIDataSource
+
+
+class PaginationConfig(BaseModel):
```

Comment:
Nit: could you make this an abstract class and create a `NoopPaginationConfig` to make it explicit that it will not paginate ?